src/razor/src/blazorDebug/blazorDebugConfigurationProvider.ts

```diff
@@ -179,6 +262,220 @@ export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurati
+            '--iCorDebugPort',
+            '-1',
+            '--OwnerPid',
+            `${process.pid}`,
+        ];
+        const spawnedProxy = cp.spawn(dotnetPath, spawnedProxyArgs);
+
+        try {
+            let chunksProcessed = 0;
+            let proxyICorDebugPort = -1;
+            let proxyBrowserPort = -1;
+            for await (const output of spawnedProxy.stdout) {
+                // If we haven't found the URL in the first ten chunks processed
+                // then bail out.
+                if (chunksProcessed++ > 10) {
+                    return ['', -1, -1];
+                }
+                BlazorDebugConfigurationProvider.vsWebAssemblyBridgeOutputChannel.appendLine(output);
+                // The debug proxy server outputs the port it is listening on in the
+                // standard output of the launched application. We need to pass this URL
+                // back to the debugger so we extract the URL from stdout using a regex.
+                // The debug proxy will not exit until killed via the `killDebugProxy`
+                // method so parsing stdout is necessary to extract the URL.
+                const matchExprProxyUrl = 'Now listening on: (?<url>.*)';
+                const matchExprICorDebugPort = 'Listening iCorDebug on: (?<port>.*)';
+                const matchExprBrowserPort = 'Waiting for browser on: (?<port>.*)';
+                const foundProxyUrl = `${output}`.match(matchExprProxyUrl);
+                const foundICorDebugPort = `${output}`.match(matchExprICorDebugPort);
+                const foundBrowserPort = `${output}`.match(matchExprBrowserPort);
+                const proxyUrlString = foundProxyUrl?.groups?.url;
+                if (foundICorDebugPort?.groups?.port != undefined) {
+                    proxyICorDebugPort = Number(foundICorDebugPort?.groups?.port);
+                }
+                if (foundBrowserPort?.groups?.port != undefined) {
+                    proxyBrowserPort = Number(foundBrowserPort?.groups?.port);
+                }
+                if (proxyUrlString) {
+                    BlazorDebugConfigurationProvider.vsWebAssemblyBridgeOutputChannel.appendLine(
+                        `Debugging proxy is running at: ${proxyUrlString}`
+                    );
+                    const oldPid = BlazorDebugConfigurationProvider.pidsByUrl.get(urlStr);
+                    if (oldPid != undefined) {
+                        process.kill(oldPid);
+                    }
+                    BlazorDebugConfigurationProvider.pidsByUrl.set(urlStr, spawnedProxy.pid);
+                    const url = new URL(proxyUrlString);
+                    return [
+                        `${url.protocol.replace(`http`, `ws`)}//${url.host}{browserInspectUriPath}`,
+                        proxyICorDebugPort,
+                        proxyBrowserPort,
+                    ];
+                }
+            }
+
+            for await (const error of spawnedProxy.stderr) {
```

Comment:
I am assuming that once we return the port, this code will not run anymore? Should we instead structure this function in such a way that we continue to monitor stderr + process exit until we are ready for the debug session to end?

Reply:
Done.